topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java

```diff
@@ -100,7 +100,7 @@ public void changeReplicationFactor(Topic topic, Handler<AsyncResult<Void>> hand
                     w.write(reassignment);
                 }
                 fut.complete(reassignmentJsonFile);
-            } catch (Exception e) {
+            } catch (IOException | InterruptedException | ExecutionException e) {
```

Comment:
Tbh it's annoying to list the possible exceptions. Why doesn't it work with just `Exception` as before?

Reply:
Spotbugs is complaining it can catch an exception which was not raised in the `try` block. 
https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rec-exception-is-caught-when-exception-is-not-thrown-rec-catch-exception

Reply:
I really hate this kind of tools. Quite often they are more useless than useful. Isn't there a way to suppress this?

Reply:
We actually want to be catching `RuntimeException` here too. Really I think this is being flagged because spotbugs doesn't understand the vertx way of handling exceptions by failing a `Future` rather than rethrowing. If you had

```
catch (Exception e) {
  throw new RuntimeException(e);
}
```
(i.e. with rethrowing) then I think spotbugs wouldn't object. It may be that we can `catch(Exception | RuntimeException e)` to avoid the warning (but we will likely get some other warning because `RuntimeException` is a subtype of `Exception`). In which case is there some annotation we can apply to _this_ method to disable this specific warning? I don't want to disable the whole warning everywhere, because we have plenty of normal code.

Reply:
> Quite often they are more useless than useful.

Sometimes yes, but I think they actually try hard to ensure that the true positive rate is higher than the false positive rate. And TBH some of the things being thrown up here, while not necessarily _wrong_, is not the best code in all the world and could well benefit from being refactored.

Reply:
I totally agree on disabling just on this method I didn't mean to disable everywhere.